src/data/repository.py

```
@@ -0,0 +1,267 @@
+"""Repository helpers for interacting with the asset relationship database."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Dict, List, Optional
```

Ruff (0.14.3)

`typing.Dict` is deprecated, use `dict` instead

(UP035)